src/services/resource.js

```diff
@@ -0,0 +1,1290 @@
+          return resources;
+        }
+        return {
+          created: [
+            ...resources.created,
+            {
+              genericId: resource.genericId,
+              resourceId: resource.resourceId,
+              sourceFields: resource.sourceFields,
+            },
+          ],
+          expanded: resources.expanded,
+        };
+      }
+
+const isExpanded = matchingFromFields.some((mff) => (
+  mff.sourceFields.length > resource.sourceFields.length &&
+  resource.sourceFields.every((l) => mff.sourceFields.includes(l))
+));
+      if (isExpanded) {
+        const expanded = resources.expanded
+          ?.find((r) => r.genericId === resource.genericId
+            && r.resourceId === resource.resourceId);
+        const matching = matchingFromFields
+          .find((r) => r.genericId === resource.genericId
+            && r.resourceId === resource.resourceId);
+        if (expanded) {
+          expanded.sourceFields = [...new Set([
+            ...expanded.sourceFields,
+            ...resource.sourceFields,
+          ])];
+          return resources;
+        }
+        return {
+          created: resources.created,
+          expanded: [
+            ...(resources.expanded || []),
+            {
+              genericId: resource.genericId,
+              resourceId: resource.resourceId,
+              sourceFields: [...new Set([
+                ...matching.sourceFields,
+                ...resource.sourceFields,
+              ])],
+            },
+          ],
+        };
+      }
+
+      return resources;
+    }, { created: [], expanded: [] });
+
+  // pull all of the removed and reduced resources in a single pass over the currentResources.
+  const removedReducedResources = currentResources
+    .reduce((resources, resource) => {
+      const isRemoved = incomingResources
+        .filter((rff) => rff.genericId === resource.genericId
+          && rff.resourceId === resource.resourceId)
+        .length === 0;
+      if (isRemoved && resource.onApprovedAR !== true) {
+        const removed = resources.removed
+          ?.find((r) => r.genericId === resource.genericId);
+        if (removed) {
+          removed.resourceIds = [...removed.resourceIds, resource.resourceId];
+          return resources;
+        }
+        return {
+          removed: [
+            ...resources.removed,
+            {
+              genericId: resource.genericId,
+              resourceIds: [resource.resourceId],
+            },
+          ],
+          reduced: resources.reduced,
+        };
+      }
+
+      const matchingFromFields = incomingResources
+        .filter((rff) => rff.genericId === resource.genericId
+        && rff.resourceId === resource.resourceId);
+      const isReduced = matchingFromFields
+        .filter((mff) => resource.sourceFields
+          .filter((l) => mff.sourceFields.includes(l))
+          .length < resource.sourceFields.length)
+        .length > 0;
```

Comment:
here as well, using `some` instead of `filter`.`length`:

```suggestion
const isReduced = matchingFromFields.some((mff) => (
  mff.sourceFields.some((l) => resource.sourceFields.includes(l)) &&
  mff.sourceFields.length < resource.sourceFields.length
));
```